Add polish translators

jakubl7545 opened the pull request:
This PR adds translators to polish speech and braille as well as tables and unit tests for both. It is compatible with Python 2 and 3 and consistent with changes recently made to the library. Test coverage for speech is at 100% and for braille is at 98% but I'm not sure if it's possible to achieve 100% for this one.

jakubl7545 committed:
Add translator to polish speech.

jakubl7545 committed:
Add translator for polish braille.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        Returns a touple with translated string and index of
+        first char after end of super.'''
+        arg=get_arg(input,start)
+		#Handle squared, degrees  and cubed as special cases
```
Since the rest of the file uses spaces for indentation this line should not be indented with tabs.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+    def sub(self,input,start,rting=()):
+        '''Translates subscripts to Polish braille.
+
+        Returns a touple, as above'''
```
While this is a copy / paste relict please correct 'touple' to tuple in both translators - that should be pretty quick with find and replace.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        else:
+            denominator=("",numerator[1],numerator[1])
+        if numerator[0].isdigit() and denominator[0].isdigit():
+            if input[start-6].isdigit() and start>5:
```
These two conditions should be swapped i.e. we should first check if `start` is bigger than 5, and only if it is try to check if there is a digit before the fraction.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        return (translation,arg[1])
+
+    def tag(self,input,start):
+        '''Translate  tags into Nemeth.
```
This still mentions Nemeth. Since I believe we don't know how tags should be translated in Polish Braille I think this method should be removed. Alternatively we may try to look up it in official notation reference, but I'm not sure if referencing equations is described there at all.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+
+        Returns a boolean.'''
+        lettersign = False # no lettersign yet
+        if input[start].lower () in 'aiouwz' and (self.lastnumber <= 0 or start != self.lastnumber) and ((start > 0 and len(input)-start>1 and input[start-1] == ' ' and input[start+1] == ' ') or (start == 0 and len(input)>1 and input[start+1] == ' ')): # special case no letter sign required, because these letter's are used as a separate words in polish
```
I understand we don't have any particular requirements with regard to line lengths, but when this condition is in a single line it is completely unreadable. Would it be possible to split it up using parents?

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        start=start-1 # We are focused on current char 
+        translation= "" # The brf translation 
+        cap = True # Provide a capital sign unless special case (below)
+        doublecap = False # Do we represent consecutive capital letters by ,,
```
Please fix this comment so that it mentions the correct symbol for consecutive capitals for Polish Braille (I believe ",," is used in UEB, as that is where we copied this code from in 2018:-)

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        translation= "" # The brf translation 
+        cap = True # Provide a capital sign unless special case (below)
+        doublecap = False # Do we represent consecutive capital letters by ,,
+        lettersign = self.letterSign (input,start) # Do we provide a lettersign (;)
```
Same here - semicolon is not a letter sign in Polish Braille.

lukaszgo1 commented on latex_access/polish_braille.py:
```diff
@@ -0,0 +1,391 @@
+        cap = True # Provide a capital sign unless special case (below)
+        doublecap = False # Do we represent consecutive capital letters by ,,
+        lettersign = self.letterSign (input,start) # Do we provide a lettersign (;)
+        try: # Letter before wasn't a cap, but letter after is so start consecutive capitals (,,)
```
And here - no ,, for consecutive capitals.

lukaszgo1 reviewed (changes requested):
Some preliminary remarks for Braille translator - I'll review rest of this code later this week.

lukaszgo1 commented on latex_access/polish_speech.py:
```diff
@@ -0,0 +1,430 @@
+                    translation = "i " + numerator[0] + denominators[int(denominator[0])]
+                else:
+                    translation = numerator[0] + denominators[int(denominator[0])]
+                if int(numerator[0]) > 1:
```
Why this code is here? Adding an empty string to a translation seems pointless.

lukaszgo1 commented on latex_access/polish_speech.py:
```diff
@@ -0,0 +1,430 @@
+    def prod(self, input, start):
+        '''Translate products, including limits of production.
+        Returns touple.
+        This is not currently translated to Polish.'''
```
This docstring is outdated - method in fact translates to Polish.

lukaszgo1 commented on tests/unit/test_polish_speech.py:
```diff
@@ -0,0 +1,117 @@
+        self.assertEqual(self.speech.super('{-4}', 0), (u' do potęgi  minus 4 ', 4))
+
+    def test_sub(self):
+        """Tests translatios of subscripts."""
```
translatios -> translations

lukaszgo1 commented on tests/unit/test_polish_speech.py:
```diff
@@ -0,0 +1,117 @@
+    @parameterized.expand([(polish_speech.speech.sum, "suma"), (polish_speech.speech.prod, "iloczyn"),
+                           (polish_speech.speech.union, "suma"), (polish_speech.speech.cap, "iloczyn")])
+    def test_other_functions(self, function, result):
+        """Tests translatios of sums, products, unions and intersections."""
```
```suggestion
        """Tests translations of sums, products, unions and intersections."""
```

lukaszgo1 committed:
Add final unit tests

lukaszgo1 committed:
Fix Py2 compat

jakubl7545 committed:
Address review.

jakubl7545 commented:
@lukaszgo1 I've addressed everything you've pointed out in the review apart from adding unit tests for % and abs as it doesn't seem to work at all. It requires further investigation into why it's not translated. Maybe in another PR.

lukaszgo1 reviewed (changes requested):
> I've addressed everything you've pointed out in the review apart from adding unit tests for % and abs as it doesn't seem to work at all. It requires further investigation into why it's not translated. Maybe in another PR.

Please add these unit tests anyway. According to my notes from our previous work on LA fixing both of these requires a single line changes to the main translators. Merging half working code is not that good if we can make it work quickly.

lukaszgo1 committed:
Add % (percent sign) to the list of symbols which are identified as LaTeX commands.

This is necessary to add translations of percents for Polish speech and Braille.

lukaszgo1 committed:
First argument of a general command can now be unicode.

Necessary to fix translations of absolute value for Polish speech and Braille.

lukaszgo1 committed:
Fix a misspell

lukaszgo1 commented:
@jakubl7545  I hope these unit tests would pass now.

lukaszgo1 committed:
Remove a condition which was always true from the method responsible for translating fractions to Braille

lukaszgo1 committed:
Merge branch 'polish_translators' of https://github.com/latex-access/latex-access into polish_translators

lukaszgo1 committed:
Simplify translations of powers

lukaszgo1 committed:
Remove unnecessary exception handling from the code which marks upper letters

lukaszgo1 committed:
Make code in Polish Braille more concise

jakubl7545 committed:
Add missing tests for abs and percent

jakubl7545 commented:
@lukaszgo1 I've finally added the missing tests.

lukaszgo1 reviewed (dismissed):
Great work! This should be merged after #29. Obviously translators would need to be moved into new packages, and Braille translator class would need to be renamed according to the new standard.

jakubl7545 committed:
Merge branch 'master' into polish_translators

jakubl7545 committed:
Apply changes made in a library.
